### PR TITLE
Renamed HEADLESS_CLIENT to DISPLAY_TYPE

### PR DESCRIPTION
**Pull request by andre-stefanov**
Renamed HEADLESS_CLIENT definition to DISPLAY_TYPE

### CODE REVIEW

File: Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp [hunk 1]
```diff
@@ -214,8 +207,8 @@
 //
 // Define some things, dont change: ///
 #define ESPBOARD
-#undef HEADLESS_CLIENT
-#define HEADLESS_CLIENT 1
+#undef DISPLAY_TYPE
+#define DISPLAY_TYPE DISPLAY_TYPE_NONE
```
Comment (ClutchplateDude) [inline Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp:211]: This is not longer the case. We can attach a display to thw ESP32 now, so we shouldn't force this.

File: Software/Arduino code/OpenAstroTracker/Constants.hpp [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-* This file contains constants that should not be changed.
+* This file contains constants that SHOULD NOT BE CHANGED by oat users!
+* If you are a developer and want to add new hardware support, add a
+* proper definition here with increased value.
 */
 
-// Supported stepper models, do not modify these values.
+// Supported display types
+#define DISPLAY_TYPE_NONE           0
+#define DISPLAY_TYPE_LCD_KEYPAD     1
```
Comment (ClutchplateDude) [inline Software/Arduino code/OpenAstroTracker/Constants.hpp:11]: I think we need more comments about what this is. Like "LCD shield on top Mega" and "I2C LCD Display" or something :-)

**Review by ClutchplateDude — CHANGES_REQUESTED** [review]

File: Software/Arduino code/OpenAstroTracker/Constants.hpp (hunk 2 above)
Comment (andre-stefanov) [reply Software/Arduino code/OpenAstroTracker/Constants.hpp:11]: Good point, i will add more descriptions

File: Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp (hunk 1 above)
Comment (andre-stefanov) [reply Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp:211]: Ah good to know, I'll remove it

**Review by ClutchplateDude — APPROVED** [review]

**Pull request merged**